x-pack/legacy/plugins/fleet/server/routes/agents/put.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Joi from 'joi';
+import { FrameworkRequest } from '../../adapters/framework/adapter_types';
+import { ReturnTypeUpdate } from '../../../common/return_types';
+import { FleetServerLib } from '../../libs/types';
+import { Agent } from '../../../common/types/domain_data';
+
+export const createPUTAgentsRoute = (libs: FleetServerLib) => ({
+  method: 'PUT',
+  path: '/api/fleet/agents/{agentId}',
+  options: {
+    tags: ['access:fleet-write'],
+    validate: {
+      payload: Joi.object({
+        user_provided_metadata: Joi.object().optional(),
```

Comment:
should this be optional? what happens if the request is sent without this payload?

Reply:
this is going to act like a PATCH request so nothing is going to be updated,